catch_analysis_tools/astrometry.py

```diff
@@ -0,0 +1,435 @@
+        res = requests.get(f"{base_url}/catch", params=params, timeout=10)
+        res.raise_for_status()
+        data = res.json()
+
+        if data.get('message') == 'Found cached data.  Retrieve from results URL.':
+            results_url = data.get('results')
+            res = requests.get(results_url, timeout=10)
+            res.raise_for_status()
+            data = res.json()
+
+        try:
+            entry = data['data'][fn]
+        except (IndexError, KeyError):
+            raise IndexError(f"Frame index {fn} out of range")
+
+        cutout_url = entry.get('cutout_url')
+        hdulist = fits.open(cutout_url)
+        image = fitsio.read(cutout_url)
+        hdulist.writeto(input_fits, overwrite=True)
+
+    return input_fits, file_base, hdulist, image
+
+
+def run_solve_field(input_fits, output_wcs, pixel_scale, scale_units="arcsecperpix"):
+    """
+    Execute the `solve-field` command to compute a WCS solution.
+
+    Parameters
+    ----------
+    input_fits : str
+        Path to the input FITS image.
+    output_wcs : str
+        Path for the output WCS solution file.
+    pixel_scale : float
+        Approximate pixel scale (e.g., arcsec/pixel).
+    scale_units : str, optional
+        Units for pixel scale (default is "arcsecperpix").
+
+    Returns
+    -------
+    success : bool
+        True if the solve-field command succeeded or file already exists.
+    """
+    if os.path.exists(output_wcs):
+        print(f"Output file '{output_wcs}' already exists. Skipping solve-field execution.")          
+        return True
+    
+    command = [
+        'solve-field',
+        '--overwrite',
+        '--scale-units', scale_units,
+        '--scale-low', str(pixel_scale * 0.5),
+        '--scale-high', str(pixel_scale * 2.0),
+        input_fits
+    ]
+    try:
+        subprocess.run(command, check=True)
+        return True
+    except subprocess.CalledProcessError as e:
+        raise RuntimeError(f"solve-field failed: {e}")
+
+
+def find_sources(image, snr, aperture_radius=7.0):
+    """
+    Detect sources in an image using SEP background subtraction and extraction.
+
+    Parameters
+    ----------
+    image : np.ndarray or MaskedArray
+        2D image data, possibly masked, for source detection.
+    snr : float
+        Minimum signal-to-noise ratio threshold for source extraction.
+    aperture_radius : float, optional
+        Radius of the circular aperture in pixels for flux summation (default is 7.0).
+
+    Returns
+    -------
+    source_list : pd.DataFrame
+        Table of detected sources with aperture photometry columns.
+    telescope_image_sub : np.ndarray
+        Background-subtracted image array.
+    """
+    mask_lower_limit = 0 
+    mask = np.isinf(image) | (image <= mask_lower_limit)
+    image = np.ma.masked_array(image, mask)
+    image_data = np.asarray(
+        getattr(image, "filled", lambda x: x)(0),
+        dtype=np.float32
+    )
+    bkg = sep.Background(image_data)
```

Comment:
We are keeping all of our background removal code in the background.py file.  So, could this be modified to assume the image is already background subtracted?